module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SsoTest.php

```diff
@@ -0,0 +1,117 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\Mink;
+
+/**
+ * Mink SSO test class.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ * @retry    4
+ */
+class SsoTest extends \VuFindTest\Integration\MinkTestCase
+{
+    /**
+     * Get config.ini override settings for testing SSO login.
+     *
+     * @return array
+     */
+    public function getConfigIniOverrides()
+    {
+        return [
+            'config' => [
+                'Authentication' => [
+                    'method' => 'SimulatedSSO',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Test changing a password.
+     *
+     * @retryCallback tearDownAfterClass
+     *
+     * @return void
+     */
+    public function testLogin()
+    {
+        // Set up configs
+        $this->changeConfigs($this->getConfigIniOverrides());
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Login
+        $this->clickCss($page, '#loginOptions a');
+
+        // Log out
+        $this->clickCss($page, '.logoutOptions a.logout');
+    }
+
+    /**
+     * SSO login in lightbox
+     *
+     * @return void
+     */
+    public function testLightboxLogin()
+    {
+        // Set up configs
+        $this->changeConfigs($this->getConfigIniOverrides());
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Record/testsample1');
+        $page = $session->getPage();
+
+        // Try to save record to list
+        $this->clickCss($page, '.record-nav .save-record');
+
+        // Login in lightbox
+        $this->clickCss($page, '.modal-body .btn.btn-link');
+
+        // Check if save form is in lightbox
+        $this->waitForPageLoad($page);
+        $this->assertLightboxTitle($page, "Add Journal of rational emotive therapy : to saved items");
+
+        // Close lightbox
+        $this->closeLightbox($page);
+
+        // Check that we are still on the record page
+        $this->assertEquals(
+            'Journal of rational emotive therapy : the journal of the Institute for Rational-Emotive Therapy.',
+            $this->findCss($page, '.record .media-body h1')->getText()
+        );
+
+        // Log out
+        $this->clickCss($page, '.logoutOptions a.logout');
```

Comment:
Not sure if logging out is strictly necessary, though it doesn't hurt. Might be worth an assertion afterward, as above. Maybe you could use a protected support method like `logOutAndAssertSuccess` to share the logic.

Reply:
Thanks for these latest changes -- looks good! You might want to consider passing $page to the logOutAndAssertSuccess method since you already have access to it in the calling context, and it will simplify the method a little bit... but no pressure to do so if you prefer it as-is.

Beyond that, I'm still waiting for feedback from others, but from my perspective, this is looking very good.

Reply:
I thought about this but decided against it. I prefer this because calling the method is simpler this way :)